pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -250,31 +254,45 @@ func doUpdateContentTree(ctx *volumemgrContext, status *types.ContentTreeStatus)
 
 		// if we made it this far, the entire tree has been verified
 		// before we mark it as verified, load it into the CAS store
-		// for now, we only load containers into containerd
-		// TODO: next stage, load disk images as well
-		if status.Format == zconfig.Format_CONTAINER {
-			loadedBlobs, err := ctx.casClient.IngestBlobsAndCreateImage(
-				getReferenceID(status.ContentID.String(), status.RelativeURL),
-				lookupBlobStatuses(ctx, status.Blobs...)...)
+
+		blobStatuses := lookupBlobStatuses(ctx, status.Blobs...)
+
+		// if we just had an image pointing to a single blob that is not index or manifest, we need
+		// to add a manifest to it.
+		if len(blobStatuses) == 1 && !blobStatuses[0].IsManifest() && !blobStatuses[0].IsIndex() {
+			blobs, err := getManifestsForBareBlob(blobStatuses[0].Path)
```

Comment:
This assumes that Path is set, but that doesn't appear to be the case when the BlobStatus is initialized from CAS after a device reboot.